src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs

```diff
@@ -36,20 +39,45 @@ private partial class AsyncSuggestedActionsSource : SuggestedActionsSource, ISug
             {
             }
 
-            public async IAsyncEnumerable<SuggestedActionSet> GetSuggestedActionsAsync(
+            public async Task GetSuggestedActionsAsync(
                 ISuggestedActionCategorySet requestedActionCategories,
                 SnapshotSpan range,
-                [EnumeratorCancellation] CancellationToken cancellationToken)
+                ImmutableArray<ISuggestedActionSetCollector> collectors,
+                CancellationToken cancellationToken)
             {
                 AssertIsForeground();
+                using var _ = ArrayBuilder<ISuggestedActionSetCollector>.GetInstance(out var completedCollectors);
+                try
+                {
+                    await GetSuggestedActionsWorkerAsync(
+                        requestedActionCategories, range, collectors, completedCollectors, cancellationToken).ConfigureAwait(false);
+                }
+                finally
+                {
+                    // Always ensure that all the collectors are marked as complete so we don't hang the UI.
+                    foreach (var collector in collectors)
```

Comment:
the idea is that you complete collector as soon as you're done giving it items.
completing all collectors at once does not give users the benefit of progressively streaming in items, and makes them high priority actions wait for the slow low priority actions to compute

Reply:
See [this sample](https://gist.github.com/AmadeusW/6ce38855a8fb14d01af228de57ecdfb8) for inspiration